lib/lbt/bundle/BundleWriter.js

```diff
@@ -20,11 +20,17 @@ class BundleWriter {
 		this.segments = [];
 		this.currentSegment = null;
 		this.currentSourceIndex = 0;
+		this.endsWithNewLine = true; // Initially we don't need a new line
 	}
 
 	write(...str) {
+		let writeBuf = "";
 		for ( let i = 0; i < str.length; i++ ) {
-			this.buf += str[i];
+			writeBuf += str[i];
+		}
+		if ( writeBuf.length >= 1 ) {
+			this.buf += writeBuf;
+			this.endsWithNewLine = ENDS_WITH_NEW_LINE.test(writeBuf);
```

Comment:
This can give wrong results when write is called with blanks or tabs only. In that case, the old endsWithNewLine flag must be taken into account as well.

Correct would be something like

```
  ENDS_WITH_NEW_LINE.test(writeBuf) || (this.endsWithNewLine && !writeBuf.trim())
```

Instead of the trim(), one could use another RegExp /^[ \t]*$/

Reply:
Sounds right 👍🏻 
As there were no dedicated unit tests, I've added one to cover the whole class.